### ISSUE TITLE
STIG Import Tool

### ISSUE THREAD

**Issue by Moe0303** [Vulnerator/Vulnerator#45]
Create functionality to import findings from STIG Checklists created with STIG Import tool.

**Comment by amkuchta** [comment]
@Moe0303 is it a safe assumption that the "STIG Import Tool" in question is the one created by Julie Hart?

**Comment by Moe0303** [comment]
yeah, I tried to upload it, but it is an .xls.  I figured you have access.

**Comment by amkuchta** [comment]
@Moe0303 I have access - I'll look at incorporating it into a future release.  As a side note, is there any particular reason that people are still using this instead of the DISA STIG Viewer?

**Comment by Moe0303** [comment]
It's a lot easier to navigate for actually doing STIG work.  The spreadsheet format makes it easy to look across several checks simultaneously vs clicking through a single check at a time.  Filters can be used to address similar checks with the same mitigation if it applies.  It is also handles the deltas more smoothly than the STIG Viewer/STIG Fusion combo.  There is no need to track down other files or open a different application.  You just re-import the new STIG on top of the old one and it will highlight the rows that were changed, added, or deleted.  It keeps all of the old information. 

It also has the ability to import and track multiple SCC outputs into the same checklist.  It will record the hostname for systems affected and set the status accordingly.  So I can run SCC on 10 systems and import all of those results into one checklist, find my outliers and address the issues.

From a reviewer/validator standpoint, it makes documenting and addressing comments a little easier since a new column can be created with reviewer comments and used to filter the results.  While the STIG viewer does have the nice export feature, I don't think it allows you to re-import back into .ckl from the .csv.

**Comment by Moe0303** [comment]
I see two advantages STIG viewer has over the STIG import tool.
1. It is able to import STIGs from a zipped file
2. It's importable into Vulnerator.

I don't care about number 1, but number 2 would be nice. :)

**Comment by Gerosolina** [comment]
The STIG Import Tool show the not reviewed items that the SCAP tool raw results generates.  In order for vulnerator to get this data you have to run everything through STIG viewer then create a .CKL file and import it to vulnerator. It would be nice if Vulnerator could take the raw SCAP .XCCFD files and show the not reviewed items in the final vulnerator output.

**Comment by Moe0303** [comment]
@gerosolina, you import Scap Compliance Checker results into a .ckl to get the not reviewed items. Then, you can import just the .ckl into the vulnerator to track the entire STIG. 

The STIG Import Tool provides the same functionality as the DISA STIG Viewer, but in a different format with added functionality.

**Comment by ibjohn** [comment]
Having long STIGs updated by DISA and having to manually transfer data from old .ckl to new .ckl sucks!  Can't wait for you to add this feature. It will help a LOT of people.

**Comment by rkotlarz** [comment]
@ibjohn, If you need a moderately fast, yet still manual process of updating older CKL files to newer ones perform the following steps.

1. Download a clear text editor / IDE such as Notepad++, Eclipse, Visual Studio
2. Create "blank" CKL files using the latest STIG Viewer from DISA.
3. Open your old CKL file with the clear text editor that you downloaded.
4. Copy all data between XML tags, <VULN> and </VULN>
5. Open the corresponding "blank" CKL file with the same clear text editor.
6. Replace all data between XML tags, <VULN> and </VULN>, with what you copied from your old CKL file. 
7. Save the file and you're done. This saved file should load into the latest STIG viewer and any tools that leverage the new format without a problem.

I wrote a PowerShell script to do this but it's been almost 2 years since I needed it so I'm not sure if I still have it or not.

**Comment by amkuchta** [comment]
@ibjohn @rkotlarz I think you guys have a different idea than what the original meaning of this issue was, but I am working on incorporating it as well.  I built another tool a while back that does a migration of data from old CKL files to new ones following the quarterly release called "STIG Fusion".  I haven't updated it recently, so I am not sure as to how well it works these days, but Iam going to put a similar functionality into the next iteration of Vulnerator.  I am currently working on revamping *a lot* of code and incorporating *a lot* of features, so bear with me!

**Comment by rkotlarz** [comment]
@amkuchta Awesome, thanks!

**Comment by CyberSecDef** [comment]
Hey guys...Just wanted to let you know a package I maintain on GitHub that offers this STIG Updating process....along with a million other things.  It's a suite of powershell scripts that make maintaining and updating a lot easier.

The repo's are attached to

https://github.com/CyberSecDef

CSTS is the live version of the suite and this particular function is covered in the updateStig.ps1 script.
I'm currently working on a new interface and new PS5 functionality which you can see in the CSTS2 repo.

Feel free to check it out.  And Feedback is always welcome.



Robert F. Weber, Jr.
Senior Cyber Security Solutions Engineer
BS:IT/SW, CASP, CEH, FQNV, MCITP: EA/SA, CIW, JSE6, DBA, A/Net/Sec+
Office 540-653-9831
Mobile 540-538-7764
robert.weber.ctr@navy.mil<mailto:robert.weber.ctr@navy.mil>
weberjr_robert@bah.com<mailto:weberjr_robert@bah.com>
http://www.boozallen.com/

"The quieter you become, the more you are able to hear."
PS > [string][char[]]("81786568818313866865688113668381317764858813767275"-split'(..)'|?{$_;}|%{33+$_;})-replace" "
________________________________

**Comment by ibjohn** [comment]
@CyberSecDef and rkotlarz  - thanks, we'll give them a shot.

**Comment by ibjohn** [comment]
@amkuchta Where can I get STIG Fusion at?

**Comment by testiando** [comment]
I agree with being able to add a STIG viewer functionality into Vulnerator. This will help IA personnel while going thru a system self validation process and also will help validators/certifiers. I see a lot of IA/Cyber-security individuals being confused on how to use the STIG viewer. 

My advise is that whatever tool is used in conjunction with Vulnerator allows any item selected as "**OPEN**" can automatically expand three additional pan windows on the right side. These three windows should include: Self-validation rational, Self-validation Mitigation and Validator Comments.  People will then know where to note particular comments making it easier for the user to know where to place these items as many are confused currently.   In addition,  IA personnel and validators can export this as spreadsheet and concentrate on the "**Open items**" rather than all items making process more efficient.  Also the spreadsheet should have a tab for IA personnel to create their deviation report based on the input already commented in the STIG viewer/tool as an automatic generated report process.

**Comment by ibjohn** [comment]
What do you think of adding Vulnerator functionality to make it a STIG management tool instead of just a reporting tool?

**Comment by rkotlarz** [comment]
Can you provide what characteristics / features you would define as a "STIG management tool"?

**Comment by ibjohn** [comment]
Off the top of my head:
-Runs as a service
-Monitors directory of .ckl files
-Keeps date of discovery for each open finding
-Emails alerts for new findings or old findings based on roles per STIG: ie: SA, NA - with all going to the ISSO/Management (recommend set up in user config per STIG so you don't get hammered with different user "preferences")
-Sends reminders to check for new .ckl files
-Adds user selected .ckl(s) and transfers data from old .ckl to new and archives old .ckl
-Schedules STIG reviews based on user set threshold

**Comment by amkuchta** [comment]
@rkotlarz @ibjohn spoiler alert - I'm already building all of this into the next release! the mindset I have is that we need to stop managing artifacts and start managing data. I've incorporated the ability to ingest the quarterly STIG releases, parsing all of the data into the back end. I'm working on CKL generation now. I've also got "Delta Analysis" built in for new releases, which will only update each individual check if it's version increases and sets a notification flag.

Short version: hang tight, it's coming!
